src/main/java/org/openpnp/logging/SystemLogger.java

```diff
@@ -42,5 +38,34 @@ public void write(byte[] buf, int off, int len) {
                 Logger.error(str);
                 break;
         }
+
+    }
+
+    @Override
+    public void write(byte[] buf, int off, int len) {
+        /*
+         * Log to tinylog as long as the Logger is available (before shutdown)
+         * As tinylog will forward all logs to the console, we do not call the super method
+         */
+        if (logToLogger) {
+            byte[] pb = new byte[len];
```

Comment:
I'm having some trouble understanding the complexity of this function. Couldn't the same thing be accomplished with code like:
```
        if (logToLogger) {
            String str = new String(buf, off, len);
            if (str.endsWith("\n")) {
                logMessage.append(str, 0, str.length() - 1);
                flushLogMessage();
            }
            else {
                logMessage.append(str);
            }
        }
        else {
            super.write(buf, off, len);
        }
```

Specifically, I don't see the reason for the new byte array, the array copy or the char by char loop of the string.

Other than this, the change looks good. I tested it with and without system output and it seems to work well, whether with your code as specified  of my change listed above.